### PR TITLE
pkg/datapath/linux/route: ensure the package compiles on macOS

### PR DESCRIPTION
**Pull request by errordeveloper**

### CODE REVIEW

**Comment by errordeveloper** [comment]
So the package is called `pkg/datapath/linux/route`, I think it would be better if it didn't have Darwin stubs and instead there would be `pkg/datapath/unimplemented/route` or something like that, and an outer wrapper package that select the implementation based on build tags. Otherwise we can just drop `linux` from the import path and do what we do now with stubs in the same package - thoughts?

**Comment by errordeveloper** [comment]
test-me-please

File: pkg/datapath/linux/route/route_darwin.go [hunk 1]
```diff
@@ -20,15 +20,25 @@ import (
...
 
 	"github.com/cilium/cilium/pkg/mtu"
+
+	"github.com/vishvananda/netlink"
 )
 
-// Replace adds or replaces the specified route if necessary. Does nothing
-// for Darwin-based builds.
+// errUnsupportedOp is a common error
+var errUnsupportedOp = fmt.Errorf("Route operations not supported on Darwnin")
```
Comment (vadorovsky) [inline pkg/datapath/linux/route/route_darwin.go]: typo: s/Darwnin/Darwin/

**Review by vadorovsky — CHANGES_REQUESTED** [review]
just one nit

**Comment by errordeveloper** [comment]
test-me-please

**Pull request merged**